lib/Containers/MerkleTree.cpp

```diff
@@ -262,17 +197,17 @@ MerkleTree<Hasher, BranchingBits>::fromBottomMostCompressed(std::string_view buf
   char const* p = buffer.data();
   char const* e = p + buffer.size();
 
-  if (p + 3 * sizeof(uint64_t) > e) {
+  if (p + sizeof(Meta) >= e) {
```

Comment:
I think for a completely empty tree `e` can be equal to `p + sizeof(Meta)`
```suggestion
  if (p + sizeof(Meta) > e) {
```